Mathlib/Order/LiminfLimsup.lean

```diff
@@ -1533,3 +1579,161 @@ theorem OrderIso.liminf_apply {γ} [ConditionallyCompleteLattice β] [Conditiona
+    (h : f.IsCoboundedUnder (· ≤ ·) u ∨ f.IsCoboundedUnder (· ≤ ·) v) :
+    f.IsCoboundedUnder (· ≤ ·) (fun a ↦ max (u a) (v a)) := by
+  rcases h with (h' | h') <;>
+  · rcases h' with ⟨b, hb⟩
+    use b
+    intro c hc
+    apply hb c
+    rw [eventually_map] at hc ⊢
+    refine hc.mono (fun _ ↦ ?_)
+    simp only [max_le_iff, and_imp]
+    intro _ _
+    trivial
+
+theorem limsup_max [ConditionallyCompleteLinearOrder β] {f : Filter α} {u v : α → β}
+    (h₁ : f.IsCoboundedUnder (· ≤ ·) u := by isBoundedDefault)
+    (h₂ : f.IsCoboundedUnder (· ≤ ·) v := by isBoundedDefault)
+    (h₃ : f.IsBoundedUnder (· ≤ ·) u := by isBoundedDefault)
+    (h₄ : f.IsBoundedUnder (· ≤ ·) v := by isBoundedDefault) :
+    limsup (fun a ↦ max (u a) (v a)) f = max (limsup u f) (limsup v f) := by
+  have bddmax := IsBoundedUnder.sup h₃ h₄
+  have cobddmax := isCoboundedUnder_le_max (v := v) (Or.inl h₁)
+  apply le_antisymm
+  · refine (limsup_le_iff cobddmax bddmax).2 (fun b hb ↦ ?_)
+    have hu := eventually_lt_of_limsup_lt (lt_of_le_of_lt (le_max_left _ _) hb) h₃
+    have hv := eventually_lt_of_limsup_lt (lt_of_le_of_lt (le_max_right _ _) hb) h₄
+    refine mem_of_superset (inter_mem hu hv) (fun _ ↦ ?_)
+    simp
+  · exact max_le (c := limsup (fun a ↦ max (u a) (v a)) f)
+      (limsup_le_limsup (eventually_of_forall (fun a : α ↦ le_max_left (u a) (v a))) h₁ bddmax)
+      (limsup_le_limsup (eventually_of_forall (fun a : α ↦ le_max_right (u a) (v a))) h₂ bddmax)
+
+theorem liminf_min [ConditionallyCompleteLinearOrder β] {f : Filter α} {u v : α → β}
+    (h₁ : f.IsCoboundedUnder (· ≥ ·) u := by isBoundedDefault)
+    (h₂ : f.IsCoboundedUnder (· ≥ ·) v := by isBoundedDefault)
+    (h₃ : f.IsBoundedUnder (· ≥ ·) u := by isBoundedDefault)
+    (h₄ : f.IsBoundedUnder (· ≥ ·) v := by isBoundedDefault) :
+    liminf (fun a ↦ min (u a) (v a)) f = min (liminf u f) (liminf v f) :=
+  limsup_max (β := βᵒᵈ) h₁ h₂ h₃ h₄
+
+open Finset
+
+theorem isBoundedUnder_le_finset_sup [LinearOrder β] [OrderBot β] {f : Filter α} {F : ι → α → β}
+    {s : Finset ι} (h : ∀ i ∈ s, f.IsBoundedUnder (· ≤ ·) (F i)) :
+    f.IsBoundedUnder (· ≤ ·) (fun a ↦ sup s F a) := by
+  choose! m hm using h
+  use sup s m
+  simp only [eventually_map] at hm ⊢
+  rw [← eventually_all_finset s] at hm
+  exact hm.mono fun _ h ↦ le_of_eq_of_le (Finset.sup_apply s F _) (sup_mono_fun h)
+
+theorem isBoundedUnder_le_finset_sup' [LinearOrder β] [Nonempty β] {f : Filter α} {F : ι → α → β}
+    {s : Finset ι} (hs : s.Nonempty) (h : ∀ i ∈ s, f.IsBoundedUnder (· ≤ ·) (F i)) :
+    f.IsBoundedUnder (· ≤ ·) (fun a ↦ sup' s hs F a) := by
+  choose! m hm using h
+  use sup' s hs m
+  simp only [eventually_map] at hm ⊢
+  rw [← eventually_all_finset s] at hm
+  refine hm.mono fun a h ↦ ?_
+  simp only [Finset.sup'_apply, sup'_le_iff]
+  exact fun i i_s ↦ le_trans (h i i_s) (le_sup' m i_s)
+
+theorem isCoboundedUnder_le_finset_sup' [LinearOrder β] {f : Filter α} {F : ι → α → β}
```

Comment:
For symmetry, what about a `sup` version for this one also?

Reply:
It is already in the file, under the name `isCobounded_le_of_bot` :)